src/main/java/tools/jackson/core/io/IOContext.java

```diff
@@ -362,4 +365,11 @@ private IllegalArgumentException wrongBuf() {
         // sanity check failed; trying to return different, smaller buffer.
         return new IllegalArgumentException("Trying to release buffer smaller than original");
     }
+
+    @Override
+    public void close() {
+        if (closed.compareAndSet(false, true)) {
+            BufferRecyclerPool.offerBufferRecycler(_bufferRecycler);
```

Comment:
If `final` definition was removed, we could perhaps use null-ness of `_bufferRecycler` itself (i.e. only offer if not null)